channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -539,12 +550,18 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
      */
     private _xhrSender(payload: string[], isAsync: boolean) {
         const xhr = new XMLHttpRequest();
-        xhr[DisabledPropertyName] = true;
-        xhr.open("POST", this._senderConfig.endpointUrl(), isAsync);
+        const endPointUrl = this._senderConfig.endpointUrl();
+        try {
+            xhr[DisabledPropertyName] = true;
+        } catch(e) {
+            // If the environment has locked down the XMLHttpRequest (preventExtensions and/or freeze), this would
```

Comment:
What sort of environments lock this down? How exactly do they lock it down? Would this be an issue for legacy SDK (we are fine to not update it, if so)?

Reply:
Salesforce using Object.preventExtensions() and Object.freeze().

Its only now becoming an issue as environments start using these (ES5) features.
I don't think we want to backport to legacy, we should just take the approach that they should upgrade the SDK version if this situation occurs.